sui/src/bin/wallet.rs

```diff
@@ -64,11 +71,40 @@ async fn try_main() -> Result<(), anyhow::Error> {
     let mut app: Command = ClientOpt::command();
     app = app.no_binary_name(false);
     let options: ClientOpt = ClientOpt::from_arg_matches(&app.get_matches())?;
+
     let wallet_conf_path = options
         .config
         .clone()
         .unwrap_or(sui_config_dir()?.join(SUI_WALLET_CONFIG));
 
+    // Prompt user for connect to gateway if config not exists.
+    if !wallet_conf_path.exists() {
+        print!(
+            "Config file [{:?}] doesn't exist, do you want to connect to a Sui Gateway [yn]?",
+            wallet_conf_path
+        );
+        if matches!(read_line(), Ok(line) if line.to_lowercase() == "y") {
+            print!("Sui Gateway Url : ");
+            let url = read_line()?;
+
+            // Check url is valid
+            HttpClientBuilder::default().build(&url)?;
+
+            let keystore_path = wallet_conf_path
+                .parent()
+                .unwrap_or(&sui_config_dir()?)
+                .join("wallet.key");
+            WalletConfig {
+                accounts: vec![],
+                keystore: KeystoreType::File(keystore_path),
```

Comment:
Will the keystore be empty? If not, can we pre-populate one here so that users do not need to run new-addresses?